etc/migration/migrateUsersToShibboleth.js

```diff
@@ -0,0 +1,352 @@
+
+        return callback(userHashes);
+    });
+
+    /*!
+     * Filter users down to those that are part of the specified tenant. Then add them to the `userHashes` array
+     *
+     * @param  {Object[]}   principalHashes     The principals to filter and aggregate
+     * @param  {Function}   callback            Will be invoked when the principals are aggregated
+     */
+    function _aggregateUsers(principalHashes, callback) {
+        log().info('Checking %s principals for tenancy', principalHashes.length);
+        _.chain(principalHashes)
+            .filter(function(principalHash) {
+                return principalHash.tenantAlias === tenantAlias && PrincipalsDAO.isUser(principalHash.principalId) && principalHash.email;
+            })
+            .each(function(userHash) {
+                log().info('Adding user %s to tenant users', userHash.email);
+                userHashes.push(userHash);
+            });
+
+        return callback();
+    }
+};
+
+/**
+ * Get the external login ids for all users in a tenancy
+ *
+ * @param  {Object[]}   userHashes            The users for a tenancy
+ * @param  {Function}   callback              Standard callback function
+ * @param  {Object[]}   callback.userHashes   The users for a tenancy with ID
+ * @api private
+ */
+var _getExternalIds = function(userHashes, callback, _userHashes) {
+    _userHashes = _userHashes || [];
+
+    if (_.isEmpty(userHashes)) {
+        return callback(_userHashes);
+    }
+
+    // Take the next principal hash from the collection
+    var principalHash = userHashes.shift();
+    _findExternalId(principalHash, function(err, userHash) {
+        if (!err && userHash) {
+            // Accumulate the return information we want
+            _userHashes.push(userHash);
+        }
+
+        // Our recursive step inside the callback for _findExternalId
+        return _getExternalIds(userHashes, callback, _userHashes);
+    });
+};
+
+/**
+ * Find the external login id for the user and add it to the user hash
+ *
+ * @param  {Object}     userHash                The user hash object
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error that occurred, if any
+ * @param  {String}     callback.userHash       The user hash object including the external id
+ * @api private
+ */
+function _findExternalId(userHash, callback) {
+    var userId = userHash.principalId;
+    Cassandra.runQuery('SELECT "loginId" FROM "AuthenticationUserLoginId" WHERE "userId" = ?', [userId], function(err, rows) {
+        if (err) {
+            log().error({'err': err}, 'Failed Cassandra query for user %s\'s loginId', userHash.email);
+            _writeErrorRow(user, 'Could not retrieve loginId for this user');
+            return callback(err);
+        }
+
+        var loginId = _.chain(rows)
+            .map(Cassandra.rowToHash)
+            .pluck('loginId')
+            .first()
+            .value();
```

Comment:
Should not, since they only have Google auth enabled. That's assuming they've never allowed local login, but can't check until staging is back up (I don't want to go run queries on production).